packages/ui/src/components/Form/OneOf/OneOfSchemaList.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Card,
+  CardBody,
+  CardTitle,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  MenuToggleElement,
+  Text,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core';
+import { FunctionComponent, PropsWithChildren, Ref, useCallback, useEffect, useState } from 'react';
+import { OneOfSchemas } from '../../../utils/get-oneof-schema-list';
+import { isDefined } from '../../../utils/is-defined';
+import { SchemaService } from '../schema.service';
+
+interface OneOfComponentProps extends PropsWithChildren {
+  name: string;
+  oneOfSchemas: OneOfSchemas[];
+  selectedSchemaName?: string;
+  onSchemaChanged: (name: string | undefined) => void;
+}
+
+export const OneOfSchemaList: FunctionComponent<OneOfComponentProps> = ({
+  name,
+  oneOfSchemas,
+  selectedSchemaName,
+  onSchemaChanged,
+  children,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const onSelect = useCallback(
+    (_event: unknown, value: string | number | undefined) => {
+      setIsOpen(false);
+      onSchemaChanged(undefined);
+      onSchemaChanged(value as string);

```

Comment:
as an example, starting with the following model:

```javascript
{
  noErrorHandler: {
    id: 1234
  }
}
```
 and later on selecting a different error handler, for instance:
```javascript
{
  refErrorHandler: {
    id: 1234
    ref: mySecretErrorHandler
  }
}
```

will end up like the following, because updating the model is done by updating individual properties.
```javascript
{
  noErrorHandler: {
    id: 1234
  },
  refErrorHandler: {
    id: 1234
    ref: mySecretErrorHandler
  }
}
```